frontend/public/components/image-stream.tsx

```diff
@@ -52,14 +53,35 @@ const ImageStreamTagsRow: React.SFC<ImageStreamTagsRowProps> = ({imageStream, sp
 
 export const ImageStreamsDetails: React.SFC<ImageStreamsDetailsProps> = ({obj: imageStream}) => {
   const imageRepository = _.get(imageStream, 'status.dockerImageRepository');
+  const publicImageRepository = _.get(imageStream, 'status.publicDockerImageRepository');
+  const imageCount = _.get(imageStream, 'status.tags.length');
   const specTagByName = _.keyBy(imageStream.spec.tags, 'name');
 
   return <div>
     <div className="co-m-pane__body">
-      <ResourceSummary resource={imageStream} showPodSelector={false} showNodeSelector={false}>
-        {imageRepository && <dt>Image Repository</dt>}
-        {imageRepository && <dd>{imageRepository}</dd>}
-      </ResourceSummary>
+      <div className="row">
+        <div className={classNames('col-sm-12', {'col-md-6': publicImageRepository})}>
+          <ResourceSummary resource={imageStream} showPodSelector={false} showNodeSelector={false}>
+            {imageRepository && <dt>Image Repository</dt>}
+            {imageRepository && <dd>{imageRepository}</dd>}
+            {publicImageRepository && <dt>Public Image Repository</dt>}
+            {publicImageRepository && <dd>{publicImageRepository}</dd>}
+            <dt>Image Count</dt>
+            <dd>{imageCount ? imageCount : 0}</dd>
+          </ResourceSummary>
+        </div>
+        {publicImageRepository &&
+          <div className="col-sm-12 col-md-6">
+            <p>To log in to the public image repository:</p>
+            <CopyToClipboard value={`docker login ${publicImageRepository} -u $(oc whoami) -p $(oc whoami -t)`} />
```

Comment:
Sorry I realize now that `publicImageRepository` contains the image name, not just the server. We might have to take this back out if we don't want to parse the string to get just server :(